pkg/operator/target_config_reconciler.go

```diff
@@ -791,6 +826,45 @@ func (c *TargetConfigReconciler) manageCertificateWebhookCR(ctx context.Context,
 	return resourceapply.ApplyUnstructuredResourceImproved(ctx, c.dynamicClient, c.eventRecorder, required, c.resourceCache, gvr, nil, nil)
 }
 
+func (c *TargetConfigReconciler) manageServiceMonitor(ctx context.Context) (*unstructured.Unstructured, bool, error) {
+
+	// Create ServiceMonitor object
+	serviceMonitor := monitoringv1.ServiceMonitor{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ServiceMonitor",
+			APIVersion: "monitoring.coreos.com/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kueue-metrics",
+			Namespace: c.operatorNamespace, // Set your namespace
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"control-plane": "controller-manager",
+				},
+			},
+			Endpoints: []monitoringv1.Endpoint{
+				{
+					Interval:        "30s",
+					Port:            "metrics", // Name of the port you want to monitor
+					Scheme:          "https",
+					BearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
+					TLSConfig: &monitoringv1.TLSConfig{
+						SafeTLSConfig: monitoringv1.SafeTLSConfig{
+							InsecureSkipVerify: ptr.To(true),
```

Comment:
Hmm. InsecureSkipVerify will skip the TLS verification.

Reply:
We have more work to enable InsecureSkipVerify of false.

I posted an upstream issue [here](https://github.com/kubernetes-sigs/kueue/issues/4377).

I can create a followup ticket for this after the upstream work is merged and carried.